std/event/loop.zig

```diff
@@ -611,79 +645,85 @@ pub const Loop = struct {
+                            @TagType(fs.Request.Msg).OpenRead => |*msg| {
+                                const flags = posix.O_LARGEFILE | posix.O_RDONLY | posix.O_CLOEXEC;
+                                msg.result = os.posixOpenC(msg.path.ptr, flags, 0);
+                            },
+                            @TagType(fs.Request.Msg).OpenRW => |*msg| {
+                                const flags = posix.O_LARGEFILE | posix.O_RDWR | posix.O_CREAT | posix.O_CLOEXEC;
+                                msg.result = os.posixOpenC(msg.path.ptr, flags, msg.mode);
+                            },
+                            @TagType(fs.Request.Msg).Close => |*msg| os.close(msg.fd),
+                            @TagType(fs.Request.Msg).WriteFile => |*msg| blk: {
+                                const flags = posix.O_LARGEFILE | posix.O_WRONLY | posix.O_CREAT |
+                                    posix.O_CLOEXEC | posix.O_TRUNC;
+                                const fd = os.posixOpenC(msg.path.ptr, flags, msg.mode) catch |err| {
+                                    msg.result = err;
+                                    break :blk;
+                                };
+                                defer os.close(fd);
+                                msg.result = os.posixWrite(fd, msg.contents);
+                            },
+                        }
+                        switch (node.data.finish) {
+                            @TagType(fs.Request.Finish).TickNode => |*tick_node| self.onNextTick(tick_node),
+                            @TagType(fs.Request.Finish).DeallocCloseOperation => |close_op| {
+                                self.allocator.destroy(close_op);
+                            },
+                            @TagType(fs.Request.Finish).NoAction => {},
+                        }
+                        self.finishOneEvent();
+                    }
+                    const rc = os.linux.futex_wait(@ptrToInt(&self.os_data.fs_queue_len), os.linux.FUTEX_WAIT, processed_count, null);
+                    switch (os.linux.getErrno(rc)) {
+                        0 => continue,
+                        posix.EINTR => continue,
+                        posix.EAGAIN => continue,
+                        else => unreachable,
+                    }
                 }
-                self.finishOneEvent();
-            }
-            const rc = os.linux.futex_wait(@ptrToInt(&self.os_data.fs_queue_len), os.linux.FUTEX_WAIT, processed_count, null);
-            switch (os.linux.getErrno(rc)) {
-                0 => continue,
-                posix.EINTR => continue,
-                posix.EAGAIN => continue,
-                else => unreachable,
-            }
+            },
+            else => @compileError("Unsupported OS"),
         }
     }
 
+    // LinuxOsData MacOsData are not directly
+    // included into the struct because the compiler cannot resolve
+    // os specific members on other platforms -- this could be a bug.
```

Comment:
@andrewrk hey, is this a bug?

Reply:
I'm still thinking about this one. We have to decide whether name resolution is lazy or not. This is actually related to #21 and I'll use the insight from implementing that issue when figuring out how to deal with this situation.